web/source/kmwbase.ts

```diff
@@ -73,17 +87,18 @@ namespace com.keyman {
     touchAliasing: dom.DOMEventHandlers;
 
     // Defines option-tracking object as a string map.
-    options: { [name: string]: string; } = {
-      'root':'',
-      'resources':'',
-      'keyboards':'',
-      'fonts':'',
-      'attachType':'',
-      'ui':null,
-      'setActiveOnRegister':'true'
+    options: OptionType = {
+      root: '',
+      resources: '',
+      keyboards: '',
+      fonts: '',
+      ui: null,
```

Comment:
Though all the fields are optional, I hesitate to remove all these empty defaults cause I noticed other parts of KMW are checking for empty strings.

I wonder if `attachType` still needs a default though because the CI unit test Attachment/Detachment is failing

Reply:
Sounds like you need to accept `''` for `attachType` and then convert internally...

Reply:
I'll do that. Does @jahorton have any to add about `OptionType`?